### PR TITLE
#270 Function Documentation Review

### LINKED ISSUE #270: Function Documentation Review
https://r-pkgs.org/man.html#roxygen-comments

- [ ]  Is the function documentation sufficient

- [ ] Clear, descriptive title.
- [ ] An `@description` tag with full description if appropriate.
- [ ] All parameters are clearly described.
- [ ] Return value is clearly described.
- [ ] Examples included, where appropriate e.g. for task type functions but not for SLF data specific functions.
- [ ] Links to other functions / packages where appropriate.

### PR DESCRIPTION
**Pull request by catherineholland1**
Closes #270 

Update and edit to function documentation

making sure all documentation is clear and descriptive 

update and link family tag

made sure all functions documented and organised within the reference

### CODE REVIEW

File: R/fix_duplicates.R [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#' Fix the West Dunbartonshire duplicates
```
Comment (Moohan) [inline R/fix_duplicates.R:1]: I like the separation of this, but could you rename the script and functions to reference homelessness, please? At the moment it's not 100% clear which dataset they refer to from the titles.

File: R/get_cohorts_paths.R [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+#' @description Get the full path to the Demographics Cohort lookup
 #'
 #' @param year The year for the cohorts extract
 #' @param ... additional arguments passed to [get_file_path()]
 #'
 #' @return The path to the Demographics Cohort lookup
 #' as a [fs::path()]
 #' @export
-#' @family file path functions
+#' @family cohort lookup
```
Comment (Moohan) [inline R/get_cohorts_paths.R:11]:
```suggestion
#' @family cohort lookup file paths
```

File: R/get_cohorts_paths.R [hunk 3]
```diff
@@ -19,16 +20,17 @@ get_demog_cohorts_path <- function(year, ...) {
...
+#' @description Get the full path to the Service Use Cohort lookup
 #'
 #' @param year The year for the cohorts extract
 #' @param ... additional arguments passed to [get_file_path()]
 #'
 #' @return The path to the Service Use Cohort lookup
 #' as a [fs::path()]
 #' @export
-#' @family file path functions
+#' @family cohort lookup
```
Comment (Moohan) [inline R/get_cohorts_paths.R:33]:
```suggestion
#' @family cohort lookup file paths
```

File: R/get_costs_paths.R [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
+#'
+#' @description Get the full Care Home costs lookup path
 #'
 #' @param ... additional arguments passed to [get_file_path()]
 #' @param update passed through [latest_update()]
 #'
 #' @return The path to the costs lookup as an [fs::path()]
 #' @export
-#' @family file path functions
+#' @family costs lookup
```
Comment (Moohan) [inline R/get_costs_paths.R:10]:
```suggestion
#' @family costs lookup file paths
```

File: R/get_costs_paths.R [hunk 5]
```diff
@@ -17,14 +19,16 @@ get_ch_costs_path <- function(..., update = NULL) {
...
+#'
+#' @description Get the full District Nursing costs lookup path
 #'
 #' @param ... additional arguments passed to  [get_file_path()]
 #' @param update passed through [latest_update()]
 #'
 #' @return The path to the costs lookup as an [fs::path()]
 #' @export
-#' @family file path functions
+#' @family costs lookup
```
Comment (Moohan) [inline R/get_costs_paths.R:31]:
```suggestion
#' @family costs lookup file paths
```

File: R/get_costs_paths.R [hunk 6]
```diff
@@ -37,14 +41,16 @@ get_dn_costs_path <- function(..., update = NULL) {
...
+#'
+#' @description Get the full GP Out of Hours costs lookup path
 #'
 #' @param ... additional arguments passed to [get_file_path()]
 #' @param update passed through [latest_update()]
 #'
 #' @return The path to the costs lookup as an [fs::path()]
 #' @export
-#' @family file path functions
+#' @family costs lookup
```
Comment (Moohan) [inline R/get_costs_paths.R:53]:
```suggestion
#' @family costs lookup file paths
```

File: R/get_costs_paths.R [hunk 7]
```diff
@@ -57,14 +63,16 @@ get_gp_ooh_costs_path <- function(..., update = NULL) {
...
+#'
+#' @description Get the full Home Care costs lookup path
 #'
 #' @param ... additional arguments passed to [get_file_path()]
 #' @param update passed through [latest_update()]
 #'
 #' @return The path to the costs lookup as an [fs::path()]
 #' @export
-#' @family file path functions
+#' @family costs lookup
```
Comment (Moohan) [inline R/get_costs_paths.R:75]:
```suggestion
#' @family costs lookup file paths
```

File: R/get_costs_paths.R [hunk 8]
```diff
@@ -77,14 +85,16 @@ get_hc_costs_path <- function(..., update = NULL) {
...
+#'
+#' @description Get the Home Care raw costs path
 #'
 #' @param ... additional arguments passed to [get_file_path()]
 #' @param update passed through [latest_update()]
 #'
 #' @return The path to the costs lookup as an [fs::path()]
 #' @export
-#' @family file path functions
+#' @family costs lookup
```
Comment (Moohan) [inline R/get_costs_paths.R:97]:
```suggestion
#' @family costs lookup file paths
```

File: R/get_it_extract_paths.R [hunk 9]
```diff
@@ -1,13 +1,14 @@
-#' Get the full path to the IT
-#' Long Term Conditions extract
+#' IT Long Term Conditions
```
Comment (Moohan) [inline R/get_it_extract_paths.R:1]: I think these types need to have 'file path' in the title for clarity.

```suggestion
#' IT Long Term Conditions file path
```

File: _pkgdown.yml [hunk 10]
```diff
@@ -1,3 +1,5 @@
+# run pkgdown::build_reference_index()to build reference index
```
Comment (Moohan) [inline _pkgdown.yml:1]:
```suggestion
# run pkgdown::build_reference_index() to build the reference index
```

**Review by Moohan — DISMISSED** [review]
Some great work here!

I think some of the function titles and `@family` tags need changing to add 'file path' or 'path' to them so it's clear that they're just returning file paths and nothing more complicated.

Now that you're using the `@family` tags in most of the descriptions could you use `has_concept()` / `lacks_concept()` in the build reference to make that simpler? [https://pkgdown.r-lib.org/reference/build_reference.html#topic-matching]()

**Comment by Moohan** [comment]
/pkgdown

Hopefully this will trigger an action to preview the pkgdown site

**Review by Moohan — COMMENTED** [review]
I'm pretty sure this is all good now, but I'll leave it to Jen to approve and merge as I've not had time to have a proper look and it's such  big PR!

**Review by Jennit07 — APPROVED** [review]
Happy with how this looks. The documentation is clear and consistent. Happy to merge :)

**Pull request merged**